### PR TITLE
Add more elaborate syntax tests

### PR DESCRIPTION
**Pull request by snejugal**

### CODE REVIEW

File: tests/syntax/lists.fail.lispf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+(
+)
```
Comment (snejugal) [inline tests/syntax/lists.fail.lispf:2]: In fail tests, each line separately is expected to fail

**Review by ilnarkhasanov — APPROVED** [review]

File: tests/syntax/numbers.fail.lispf [hunk 2]
```diff
@@ -0,0 +1,19 @@
+9223372036854775808
```
Comment (fedor-ivn) [inline tests/syntax/numbers.fail.lispf:1]: why does it fail?

File: tests/syntax/quote.lispf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+'x 'X'Y 'func
```
Comment (fedor-ivn) [inline tests/syntax/quote.lispf:1]: > 'X'Y

Does this expression mean anything?

**Review by fedor-ivn — COMMENTED** [review]
Awesome

File: tests/syntax/numbers.fail.lispf (hunk 2 above)
Comment (snejugal) [reply tests/syntax/numbers.fail.lispf:1]: because it doesn't fit into a `i64`

File: tests/syntax/quote.lispf (hunk 3 above)
Comment (snejugal) [reply tests/syntax/quote.lispf:1]: `'X'Y` parses the same as `'X 'Y`

**Review by fedor-ivn — APPROVED** [review]

**Pull request merged**